codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/AddAwsConfigFields.java

```diff
@@ -88,6 +89,26 @@
                     .build()
     );
 
+    /**
+     * Returns false if a service has a customization required for default assigned HTTP client
+     * that is created from Client Constructor.
+     *
+     * @param model the model used for generation.
+     * @param service the service shape for which default HTTP Client is generated.
+     * @return boolean indication whether we should resolve a default AWS HTTP Buildable Client.
+     */
+    private static boolean resolveDefaultAWSHTTPBuildableClient(Model model, ServiceShape service) {
+        String serviceId= service.expectTrait(ServiceTrait.class).getSdkId();
+        // services that require modified HTTP Client
+        String []serviceList = {"S3"};
+        for (String s : serviceList) {
+            if (serviceId.equalsIgnoreCase(s)) {
+                return false;
+            }
+        }
+        return true;
+    }
```

Comment:
Is there anyway this can be triggered by GoIntegrations?  sprinkling the customizations throughout the generator code will make it difficult to track down and understand what all is being done.

Reply:
This was not needed any more, so removed.